### PR TITLE
docs: Update `dev` branch with changes on `master`

### PR DESCRIPTION
**Pull request by robin-owen**
Update the state of the documentation on the `dev` branch with the changes from `master`. This will allow us to document in-development features on the appropriate branch without generating merge conflicts in the future.

No changes were made to the content. This is simply a collection of cherry-picked commits.

### CODE REVIEW

File: docs/source/topics/ref_minimum-system-requirements.adoc [hunk 1]
```diff
@@ -12,6 +12,9 @@
 * 9 GB of free memory
 * 35 GB of storage space
 
+{prod} is supported only on AMD64 and Intel 64 processor architectures.
+{prod} does not support the ARM-based M1 architecture.
```
Comment (praveenkumar) [inline docs/source/topics/ref_minimum-system-requirements.adoc:16]: With dev branch we are trying to have M1 support, we even published a dev-preview binary to support it.

**Comment by robin-owen** [comment]
Just switched the base branch from `dev` to `main`, the changes should still apply cleanly.

File: docs/source/topics/ref_minimum-system-requirements.adoc (hunk 1 above)
Comment (robin-owen) [reply docs/source/topics/ref_minimum-system-requirements.adoc:16]: Let's create a docs issue for that and I'll address it with another PR. :)

**Comment by gbraad** [comment]
@praveenkumar this is the one that needs to be merged.

**Review by anjannath — APPROVED** [review]

**Pull request merged**